docs/Coiote_IoT_DM/firmware_update.md

```diff
@@ -0,0 +1,188 @@
+
+
+## Prepare the Firmware Update
+
+1. Select the device you want to update from the [**Coiote Device Inventory**](https://eu.iot.avsystem.cloud/ui/device/inventory).
+    
+1. Select **Data model** from the top menu and validate if the Firmware Update object `/5` is visible, indicating that the Object is supported by the LwM2M Client.
+
+    ![Firmware update object](images/State 0 - fota object.png)
+
+1. Select **Firmware update** from the top menu and click **Update Firmware** from the top-right corner.
+
+    ![Update Firmware Tab](images/Device Inventory2.png)
+
+1. Select **Basic Firmware Update**
+
+    ![Basic Firmware Update](images/Basic Firmware Update.png)
+
+1. Choose between **Pull** and **Push**
+
+    **Pull method** (recommended): The LwM2M Client receives the URI of the file that is to be downloaded and pulls the file from it. 
+    
+    **Push method**: The LwM2M Server pushes the firmware file to the device.
+
+
+    !!! Info
+        **Pull** supports any of the following **transport types**:
+
+        - `CoAP` or `CoAPs` over `UDP`
+        - `CoAP` or `CoAPs` over `TCP`
+        - `HTTP` or `HTTPs`
+
+        **Push** uses `CoAPs` over `UDP` by default.
+
+    !!! Note
+        **Which transport protocol to choose?**
+        
+        Downloads using `CoAP(s)` over `UDP` tend to be slow due to limitation of the maximum CoAP Block size of 1024 bytes and the required acknowledgements for each Block transfer.
+
+        Choosing `CoAP(s)` over `TCP` or `HTTP(s)` usually results in faster download speeds.
+
+
+    ![Basic Firmware Update](images/Pull.png)
+
+
+1. Click **Schedule Update** from the bottom-right corner to trigger the Firmware Update process.
+
+    ![Schedule Update](images/schedule.png)
+
+
+## Download & Upgrade Process
+
+If the Firmware Update is scheduled successfully, the device starts **downloading** the firmware at the next practical opportunity. The actual firmware **update** starts once the integrity and authenticity of the firmware image has been validated by the LwM2M Client.
+
+![Upgrading](images/update-in-progress.png)
+
+The status of the firmware update is shown in the Resources: **State** `/5/*/3` & **Update Result** `/5/*/5`. If no errors arise, the update process follows the following pattern:
+
+1. **Downloading** `state 1` & `update result 0` 
+2. **Downloaded** `state 2` & `update result 0` 
+3. **Updating** `state 3` & `update result 0` 
+4. **Updated** `state 0` & `update result 1` 
+
+
+!!! info
+    When the device is updating its firmware, it will deregister and reboot using the new firmware. This process may time multiple minutes.
+
+    ![Deregistered while upgrading](images/deregistered.png)
+
+
+If the update was successfully performed, the **State** `/5/*/3` returns to `0` and the **Update Results** `/5/*/5` reports `1`.
+
+![Firmware upgraded](images/result1.png)
```

Comment:
It is a little bit unobvious that your example is switching between the "Firmware update" and "Data model" tabs of the UI. You might consider either removing this part, or adding some explicit explanation (e.g. something along the lines of "During and after the upgrade process, you can see the raw State and Result resource values on the 'Data model' tab").